src/Microsoft.ML.Transforms/Text/NgramTransform.cs

```diff
@@ -178,350 +244,115 @@ public void Save(ModelSaveContext ctx)
                 Contracts.Assert(0 <= SkipLength && SkipLength <= NgramBufferBuilder.MaxSkipNgramLength);
                 Contracts.Assert(NgramLength + SkipLength <= NgramBufferBuilder.MaxSkipNgramLength);
                 ctx.Writer.Write(SkipLength);
-                Contracts.Assert(Enum.IsDefined(typeof(WeightingCriteria), Weighting));
+                Contracts.Assert(Enum.IsDefined(typeof(NgramEstimator.WeightingCriteria), Weighting));
                 ctx.Writer.Write((int)Weighting);
                 Contracts.Assert(Utils.Size(NonEmptyLevels) == NgramLength);
                 ctx.Writer.WriteBoolBytesNoCount(NonEmptyLevels);
             }
         }
 
-        private const uint VerTfIdfSupported = 0x00010002;
-
-        public const string LoaderSignature = "NgramTransform";
-        internal const string Summary = "Produces a bag of counts of ngrams (sequences of consecutive values of length 1-n) in a given vector of keys. "
-            + "It does so by building a dictionary of ngrams and using the id in the dictionary as the index in the bag.";
+        private readonly TransformInfo[] _transformInfos;
```

Comment:
>TransformInfo [](start = 25, length = 13)

can we make this field a public immutable array?

Reply:
immutable array, yes, public - why?

---
In reply to: [233688540](https://github.com/dotnet/machinelearning/pull/1592#discussion_r233688540) [](ancestors = 233688540)